### PR TITLE
feat: use custom service for screenshots

### PR DESCRIPTION
**Pull request by oroce**
example test:

```
describe('screenshot', () => {
  it('dropdown', () => {
    browser.url('/buttons');

    browser.saveElementScreenshot('.dropdown', './dropdown.png');
  });
});
```

### CODE REVIEW

File: e2e/test/buttons-tests/buttons-test.js [hunk 1]
```diff
@@ -30,6 +30,15 @@ describe('FEF buttons tests', () => {
        '"Yeah dropdown" is not similar to the reference');
   });
 
+  it.only('should check the button: ' + YEAH_DROPDOWN_NAME, () => {
```
Comment (oroce) [inline e2e/test/buttons-tests/buttons-test.js]: don't forget to remote the `.only`

File: e2e/utils/visual-regression.js [hunk 2]
```diff
@@ -174,3 +194,70 @@ module.exports.getRefPicName = () => {
...
+  }
+
+  if (options.ffAndWindowsTolerance
+  && browser.desiredCapabilities.browserName
+  && browser.desiredCapabilities.browserName.includes('firefox')) {
+    misMatchTolerance = options.ffAndWindowsTolerance;
+  }
+
+  browser.saveElementScreenshot(selector, testPathAndName);
+  if (options.ignoreComparison === true) {
```
Comment (oroce) [inline e2e/utils/visual-regression.js]: what you can do:

```
const img = resemble(getRefPicName()).compareTo(testPathAndName)
if (options.ignoreComparison === true) {
  img.ignoreColors();
}
.onComplete(function (data) {
   return handleTakenScreenshot(data, misMatchTolerance, selector);
});
```

File: e2e/utils/visual-regression.js [hunk 3]
```diff
@@ -17,11 +20,36 @@ function getMisMatchPercentage(results, testName) {
...
-  basePathDiff = dirPath + '/pics/diff';
-  basePathScreen = dirPath + '/pics/screen';
+  basePathRef = dirPath + '/pics/reference/';
+  basePathDiff = dirPath + '/pics/diff/';
+  basePathScreen = dirPath + '/pics/screen/';
 }
 
+function setBasePath2(dirPath) {
+  if (process.env.CIRCLE_ARTIFACTS) {
+    basePathDiff = process.env.E2E_SCREENSHOTS + 'dif/';
```
Comment (oroce) [inline e2e/utils/visual-regression.js]: `dif` and not `diff`?

File: e2e/utils/visual-regression.js [hunk 4]
```diff
@@ -17,11 +20,36 @@ function getMisMatchPercentage(results, testName) {
...
+  } else {
+    basePathDiff = dirPath + '/pics/diff/';
+    basePathScreen = dirPath + '/pics/screen/';
+  }
+  basePathRef = dirPath + '/pics/reference/';
+}
+
+function createTestName() {
+  const testName = browser.currentTestName;
+  const browserVersion = parseInt(browser.desiredCapabilities.version, 10);
```
Comment (oroce) [inline e2e/utils/visual-regression.js]: won't this be `NaN` for version `latest`?

File: e2e/utils/visual-regression.js [hunk 5]
```diff
@@ -174,3 +202,95 @@ module.exports.getRefPicName = () => {
...
+  }
+
+  const promise = writeDataIfNeeded(img, misMatchTolerance, selector).then(result => {
+    return {
+      value: result
+    };
+  });
+  return browser.waitUntil(() => {
+    return promise;
+  }, 1e4);
```
Comment (oroce) [inline e2e/utils/visual-regression.js]: shall we keep this 1e4? Is it the right amount of time to wait?

File: e2e/test/buttons-tests/buttons-test.js [hunk 6]
```diff
@@ -40,9 +51,9 @@ describe('FEF buttons tests', () => {
     browser.click(SAVE_BUTTON_XPATH);
 
     $(SAVE_BUTTON_SELECTOR).waitForExist(8500);
-    assert(takeScreenShotOfElement(SAVE_BUTTON_XPATH,
-      {defaultTolerance: 17.5, ignoreComparison: true}),
-       'Save button is not similar to the reference after click');
+    takeScreenShotOfElement(SAVE_BUTTON_XPATH,
```
Comment (oroce) [inline e2e/test/buttons-tests/buttons-test.js]: missing assert?

File: e2e/test/buttons-tests/buttons-test.js [hunk 7]
```diff
@@ -30,6 +31,16 @@ describe('FEF buttons tests', () => {
        '"Yeah dropdown" is not similar to the reference');
   });
 
+  it('should check the button with mouseover: ' + YEAH_DROPDOWN_NAME, () => {
+    browser.url('/buttons');
+    $(SAVE_BUTTON_SELECTOR).scroll();
+    browser.moveToElement('.dropdown');
```
Comment (oroce) [inline e2e/test/buttons-tests/buttons-test.js]: it's hard to follow this test it uses `YEAH_DROPDOWN_NAME`, `SAVE_BUTTON_SELECTOR` and even the selector of `.dropdown`

File: e2e/utils/visual-regression.js (hunk 5 above)
Comment (molnosikristof) [reply e2e/utils/visual-regression.js]: I think we can increase it to 1e5, it should be enough.

File: e2e/utils/visual-regression.js [hunk 8]
```diff
@@ -292,5 +292,5 @@ module.exports.takeScreenShotOfElementWithCrop = (selector, options) => {
   });
   return browser.waitUntil(() => {
     return promise;
-  }, 1e4);
+  }, 1e5);
```
Comment (oroce) [inline e2e/utils/visual-regression.js]: 1min40? i thought 1e4 is just too much:D

can we keep then 1e4 and increase to 20000 if necessary?

File: package.json [hunk 9]
```diff
@@ -84,6 +85,8 @@
     "classnames": "^2.2.5",
     "debug": "^3.1.0",
     "lodash.throttle": "^4.1.1",
+    "mkdirp": "^0.5.1",
```
Comment (oroce) [inline package.json]: @molnosikristof can you remove these from here and put into `devDependencies`?

File: e2e/utils/visual-regression.js (hunk 8 above)
Comment (molnosikristof) [reply e2e/utils/visual-regression.js]: No, not necessary, I think it does not matter.

File: package.json (hunk 9 above)
Comment (molnosikristof) [reply package.json]: Done

Comment (oroce) [reply package.json]: and also the `node-resemble-js` pls

File: e2e/utils/visual-regression.js (hunk 5 above)
Comment (oroce) [reply e2e/utils/visual-regression.js]: let's leave this, 10 sec should more than enough.
